[Joy UI] Add `CircularProgress` component (#33869)
mui/material-ui: docs/data/joy/components/circular-progress/circular-progress.md

---
product: joy-ui
title: React Circular Progress component
githubLabel: 'component: CircularProgress'
---

# Circular Progress

<p class="description">Circular Progress indicators, commonly known as spinners, express an unspecified wait time or display the length of a process.</p>

## Introduction

Progress indicators inform users about the status of ongoing processes, such as loading an app, submitting a form, or saving updates.

> By default, the `CircularProgress` is indeterminate which indicates an unspecified wait time. You can switch to [determinate](#determinate) mode to display how long an operation will take.

Reviewer: ```suggestion
The `CircularProgress` is indeterminate by default, indicating an unspecified wait time. 
To actually have it represent how long an operation will take, use the [determinate](#determinate) mode.
```

A bit of a rephrasing!
